exporter/datasetexporter/config_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasetexporter
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+type SuiteConfig struct {
+	suite.Suite
+}
+
+func (s *SuiteConfig) SetupTest() {
+	os.Clearenv()
+}
+
+func TestSuiteConfig(t *testing.T) {
+	suite.Run(t, new(SuiteConfig))
+}
+
+func (s *SuiteConfig) TestConfigUnmarshalUnknownAttributes() {
+	config := Config{}
+	conf := confmap.NewFromStringMap(map[string]interface{}{
+		"dataset_url":       "https://example.com",
+		"api_key":           "secret",
+		"unknown_attribute": "some value",
+	})
+
+	err := config.Unmarshal(conf)
+	s.NotNil(err)
+
+	unmarshalErr := fmt.Errorf("1 error(s) decoding:\n\n* '' has invalid keys: unknown_attribute")
+	expectedError := fmt.Errorf("cannot unmarshal config: %w", unmarshalErr)
+
+	s.Equal(err.Error(), expectedError.Error())
+	// TODO: Why it does not work?
+	// s.EqualErrorIs(err, expectedError)
```

Comment:
```suggestion
```
I'd say you can just remove this check. You could maybe use `EqualError` or `EqualValues` for the assertion above, but that's mostly stylistic. I think what you're already checking works.